File: src/pcworld/model/CPU.java
package pcworld.model;

public class CPU extends Components {
	public Components comp;
	public int id;
	public String socket;
	public float base_clock;
	public float max_clock;
	public int num_core;
	public int multithread;
	public int wattage;
	
	public CPU() {
		super();
	}

	public CPU(int id, String socket, float base_clock, float max_clock, int num_core, int multithread, int wattage) {
		super();
		this.id = id;
		this.socket = socket;
		this.base_clock = base_clock;
		this.max_clock = max_clock;
		this.num_core = num_core;
		this.multithread = multithread;
		this.wattage = wattage;
	}
	
	public CPU(int id, String brand, String model, double price, String image, String type, String socket, float base_clock, float max_clock, int num_core, int multithread, int wattage) {
		super(id, brand, model, price, image, type);
		this.socket = socket;
		this.base_clock = base_clock;
		this.max_clock = max_clock;
		this.num_core = num_core;
		this.multithread = multithread;
		this.wattage = wattage;
	}
	
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getSocket() {
		return socket;
	}

	public void setSocket(String socket) {
		this.socket = socket;
	}
	
	public float getBase_clock() {
		return base_clock;
	}

	public void setBase_clock(float base_clock) {
		this.base_clock = base_clock;
	}

	public float getMax_clock() {
		return max_clock;
	}

	public void setMax_clock(float max_clock) {
		this.max_clock = max_clock;
	}

	public int getNum_core() {
		return num_core;
	}

	public void setNum_core(int num_core) {
		this.num_core = num_core;
	}
	
	public int getMultithread() {
		return multithread;
	}

	public void setMultithread(int multithread) {
		this.multithread = multithread;
	}

	public int getWattage() {
		return wattage;
	}

	public void setWattage(int wattage) {
		this.wattage = wattage;
	}
}

File: webScraping/cpu.py
# import libraries
import MySQLdb
import requests
from selenium import webdriver
from bs4 import BeautifulSoup
import time
import re

HOST = "localhost"
USERNAME = "root"
PASSWORD = ""
DATABASE = "pcworld"

# Open database dbection
db = MySQLdb.connect(HOST, USERNAME, PASSWORD, DATABASE)

# initialize Selenium to work with Google Chrome
driver = webdriver.Chrome('./venv/Lib/site-packages/chromedriver.exe')
#url to be parsed
driver.get('https://pcpartpicker.com/products/cpu/#k=30')
# wait 1 second for web page to load before start parsing
time.sleep(1)
print('start')
page_source = driver.page_source

#parse the data
soup = BeautifulSoup(page_source, "html.parser")

# items = soup.find_all(class_=re.compile("td__spec--[1-6]"))
items = soup.find_all(class_="tr__product")
for item in items:
    img = item.find(class_='td__image')
    img1 = img.img['src']
    print(img1)
    
    name = item.find(class_='td__nameWrapper')
    name1 = name.p.text
    brand, *name2 = name1.split(" ",1)
    model = " ".join(name2)
    print("Brand : " + brand)
    print("Model : " + model)

    core = item.find(class_='td__spec--1')
    core1 = int(core.contents[1])
    print(core1)
    
    base_clock = item.find(class_='td__spec--2')
    base_clockX = base_clock.contents[1].split()
    base_clock1 = float(base_clockX[0])
    print(base_clock1)

    try:
        max_clock = item.find(class_='td__spec--3')
        max_clockX = max_clock.contents[1].split()
        max_clock1 = float(max_clockX[0])
        print(max_clock1)
    except:
        continue
    
    watt = item.find(class_='td__spec--4')
    wattX = watt.contents[1].split()
    watt1 = int(wattX[0])
    print(watt1)
    
    multi = item.find(class_='td__spec--6')
    multiX = multi.contents[1]
    if multiX == 'Yes':
        multi1 = 1
    else:
        multi1 = 0
    print(multi1)

    price = item.find(class_='td__price')
    pricetemp = price.find(text=re.compile("^\$"))
    if pricetemp==None:
        continue
    price1 = float(pricetemp.strip('$'))*4
    print(price1)

    cursor = db.cursor()
    sql1 = "insert into components (brand, model, price, image, type) VALUES (%s,%s,%s,%s,%s)"
    param1 = (brand, model, price1, img1, "CPU")
    sql2 = "insert into cpu (id, socket, base_clock, max_clock, num_core, multithread, wattage) VALUES(%s,%s,%s,%s,%s,%s,%s)"
    
    try:
        cursor.execute(sql1, param1)
        last_id = cursor.lastrowid
        param2 = (last_id, "LGA1151", base_clock1, max_clock1, core1, multi1, watt1)
        cursor.execute(sql2, param2)
        db.commit()
    except MySQLdb.Error as err:
        print(err)